pgcli/pgexecute.py

```diff
@@ -307,6 +308,12 @@ def run(self, statement, pgspecial=None, exception_formatter=None,
                 if not on_error_resume:
                     break
 
+    def _cannot_catch(self, e):
```

Comment:
I think that `_cannot_catch` is not a very descriptive name. We actually can catch the exception here, we just prefer not to. I would rather use something like `_must_raise` or `_should_raise`.